test/unit/test_manifest_download.py

```diff
@@ -0,0 +1,86 @@
+import errno
+import os
+import shutil
+import tempfile
+import unittest
+
+import scandir
+from mock import patch
+
+from hca.dss import DSSClient
+
+
+def _touch_file(path):
+    try:
+        os.makedirs(os.path.split(path)[0])
+    except OSError as e:
+        if e.errno == errno.EEXIST:
+            pass
+    with open(path, 'w'):
+        pass
+
+
+def _fake_download(*args, **kwargs):
+    _touch_file(args[2])
+
+
+class TestManifestDownload(unittest.TestCase):
+    manifest = [
+        ['bundle_uuid', 'bundle_version', 'file_content_type', 'file_name', 'file_sha256', 'file_size', 'file_uuid', 'file_version', 'file_indexed'],
+        ['a_uuid', '1_version', 'somestuff', 'a_file_name', 'ad3fc1e4898e0bce096be5151964a81929dbd2a92bd5ed56a39a8e133053831d', '12', 'af_uuid', 'af_version', 'False'],
+        ['b_uuid', '1_version', 'somestuff', 'b_file_name', '8f35071eaeedd9d6f575a8b0f291daeac4c1dfdfa133b5c561232a00bf18c4b4', '2', 'bf_uuid', 'af_version', 'False'],
+        ['c_uuid', '1_version', 'somestuff', 'c_file_name', '8f3404db04bdede03e9128a4b48599d0ecde5b2e58ed9ce52ce84c3d54a3429c', '41', 'cf_uuid', 'af_version', 'False'],
+    ]
+
+    def setUp(self):
+        self.prev_wd = os.getcwd()
+        self.tmp_dir = tempfile.mkdtemp()
+        os.chdir(self.tmp_dir)
+        self.dss = DSSClient()
+        with open('manifest.tsv', 'w') as f:
+            f.write('\n'.join(['\t'.join(row) for row in self.manifest]))
+
+    def tearDown(self):
+        os.chdir(self.prev_wd)
+        shutil.rmtree(self.tmp_dir)
+
+    def _assert_all_files_downloaded(self):
+        files_present = {os.path.join(dir_path, f) for dir_path, _, files in scandir.walk('.') for f in files}
+        files_expected = {
+            '.' + os.sep + 'manifest.tsv',
+            '.' + os.sep + 'ad' + os.sep + '3fc1' + os.sep + 'ad3fc1e4898e0bce096be5151964a81929dbd2a92bd5ed56a39a8e133053831d',
+            '.' + os.sep + '8f' + os.sep + '3507' + os.sep + '8f35071eaeedd9d6f575a8b0f291daeac4c1dfdfa133b5c561232a00bf18c4b4',
+            '.' + os.sep + '8f' + os.sep + '3404' + os.sep + '8f3404db04bdede03e9128a4b48599d0ecde5b2e58ed9ce52ce84c3d54a3429c'
+        }
```

Comment:
* Nested comprehension become harder to read as one liners.
* `os.path.join()` accepts `*args`
* `assertSetEquals` is [dicouraged](https://docs.python.org/3/library/unittest.html#unittest.TestCase.addTypeEqualityFunc)

> The list of type-specific methods automatically used by assertEqual() are summarized in the following table. Note that it’s usually not necessary to invoke these methods directly.

So:

```diff
@@ -45,14 +55,16 @@
         shutil.rmtree(self.tmp_dir)
 
     def _assert_all_files_downloaded(self):
-        files_present = {os.path.join(dir_path, f) for dir_path, _, files in scandir.walk('.') for f in files}
+        files_present = {os.path.join(dir_path, f)
+                         for dir_path, _, files in scandir.walk('.')
+                         for f in files}
         files_expected = {
-            '.' + os.sep + 'manifest.tsv',
-            '.' + os.sep + 'ad' + os.sep + '3fc1' + os.sep + 'ad3fc1e4898e0bce096be5151964a81929dbd2a92bd5ed56a39a8e133053831d',
-            '.' + os.sep + '8f' + os.sep + '3507' + os.sep + '8f35071eaeedd9d6f575a8b0f291daeac4c1dfdfa133b5c561232a00bf18c4b4',
-            '.' + os.sep + '8f' + os.sep + '3404' + os.sep + '8f3404db04bdede03e9128a4b48599d0ecde5b2e58ed9ce52ce84c3d54a3429c'
+            os.path.join('.', 'manifest.tsv'),
+            os.path.join('.', 'ad', '3fc1', 'ad3fc1e4898e0bce096be5151964a81929dbd2a92bd5ed56a39a8e133053831d'),
+            os.path.join('.', '8f', '3507', '8f35071eaeedd9d6f575a8b0f291daeac4c1dfdfa133b5c561232a00bf18c4b4'),
+            os.path.join('.', '8f', '3404', '8f3404db04bdede03e9128a4b48599d0ecde5b2e58ed9ce52ce84c3d54a3429c')
         }
-        self.assertSetEqual(files_present, files_expected)
+        self.assertEqual(files_present, files_expected)
 
     @patch('hca.dss.DSSClient.DIRECTORY_NAME_LENGTHS', [1, 2, 3])
     def test_file_path(self):
```

Reply:
Also duh. This is much better. I forgot it took `*args`. Good to know about `assertSetEquals`.